Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -473,17 +475,17 @@ class BrowserViewController: UIViewController,
         }
     }
 
-    func newState(state: FakespotState) {
+    func newState(state: BrowserViewControllerState) {
         ensureMainThread { [weak self] in
             guard let self else { return }
 
-            fakespotState = state
+            browserViewControllerState = state
 
             // opens or close sidebar/bottom sheet to match the saved state
-            if state.isOpenOnProductPage {
+            if state.fakespotState.isOpenOnProductPage {
                 guard let productURL = urlBar.currentURL else { return }
                 handleFakespotFlow(productURL: productURL)
-            } else {
+            } else if !state.fakespotState.isOpenOnProductPage {
                 _ = dismissFakespotIfNeeded()
```

Comment:
This was an `else` before, but now an `else if` - just wanna confirm this is ok?